guides/source/layout.html.erb

```diff
@@ -94,48 +106,54 @@
       </nav>
     </div>
   </header>
+
   <hr class="hide" />
 
-  <section id="feature">
-    <div class="wrapper">
-      <%= yield :header_section %>
+  <main id="container">
+    <article>
+      <header id="feature">
+        <div class="wrapper">
+          <%= yield :header_section %>
 
-      <%= yield :index_section %>
+          <%= yield :index_section %>
+        </div>
+      </header>
 
-      <hr>
-    </div>
-  </section>
+      <div class="wrapper">
+        <div id="mainCol">
+          <section id="article-body">
+            <%= yield %>
+          </section>
 
-  <main id="container">
-    <div class="wrapper">
-      <div id="mainCol">
-        <%= yield %>
-        <hr>
-        <h3>Feedback</h3>
-        <p>
-          You're encouraged to help improve the quality of this guide.
-        </p>
-        <p>
-          Please contribute if you see any typos or factual errors.
-          To get started, you can read our <%= link_to 'documentation contributions', 'https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#contributing-to-the-rails-documentation' %> section.
-        </p>
-        <p>
-          You may also find incomplete content or stuff that is not up to date.
-          Please do add any missing documentation for main. Make sure to check
-          <%= link_to 'Edge Guides', 'https://edgeguides.rubyonrails.org' %> first to verify
-          if the issues are already fixed or not on the main branch.
-          Check the <%= link_to 'Ruby on Rails Guides Guidelines', 'ruby_on_rails_guides_guidelines.html' %>
-          for style and conventions.
-        </p>
-        <p>
-          If for whatever reason you spot something to fix but cannot patch it yourself, please
-          <%= link_to 'open an issue', 'https://github.com/rails/rails/issues' %>.
-        </p>
-        <p>And last but not least, any kind of discussion regarding Ruby on Rails
-          documentation is very welcome on the <%= link_to 'official Ruby on Rails Forum', 'https://discuss.rubyonrails.org/c/rubyonrails-docs' %>.
-        </p>
+          <hr>
+
+          <footer aria-labelledby="heading-feedback">
```

Comment:
In html 5, footer may also be an element inside an article or section. I experience the same behaviour as you (because I also use NVDA when I use a screen reader) but I believe jaws does denote it differently.

Reply:
Oh, I don't know. I'm not too convinced about adding so many additional regions beyond the navigation, main region and footer. It adds a lot of noise that is already covered by the heading levels and becomes a bit repetitive information.

By the way, I personally don't tend to use many links to skip to specific sections and hope this can be done natively. And really, there's no way to skip to the body of the article with a command to go to a region. You can easily use 2, but that makes me think we're not being very consistent with using regions, or if we're even using them properly.

What do you think?

Reply:
My personal thoughts here were:

- I am browsing the guides extensively
- I am in the middle of the article and find something I want to give feedback on
- I know there is a _region_ (labelled section) that explains how to give feedback
- I want to go there straight away

With NVDA I can get there right away when browsing the article pressing <kbd>D</kbd>, which should navigate to the next landmark or region. In this case the footer is nested inside article, and thus not _complementary_, but it is labelled by the heading, and thus it becomes a section with role="region", iirc. Using the TOC navigation here would be sub-optimal as it doesn't allow be to go to "last heading of the current level", by pressing <kbd>2</kbd>. I would need to press it repeatedly, right? Since there are so little regions, that's not the case.

Does this make sense?

Reply:
We have a region for the article: https://github.com/rails/rails/pull/51499/files/93f9819c62890301f398e752b6083c9d2c9f6c40#diff-dc279d925a05373ac5f34307441799f98ac4accbe0b0a8f12f99f703585afa7dR113 Is that not working for you? I haven't tested mine but I can.

It _should_ be:

```html
<main>
  <article>

    <header id="feature">
      article introduction

      chapter index / toc
    </header>   

    article contents with lots of headings    

    <footer>
     <h2>Feedback</h2>

     feedback explanation
    </footer>
  </article>
</main>
```

Reply:
(sidenote: I am fine with explicit role-ing things if SRs don't recognise it, so let's do that if we must)

Reply:
Right! So I think assigning the explicit roles would be enough to solve this.

In any case, an article is not exposed as a region, it's exposed as another type of element. In fact, at least in NVDA, there's no default shortcut to jump to it, I configured the A to do so but it's not the common case.

Reply:
Cool! Yeah, I _love_ different support for different SRs and even different versions of the same SR 🗡️ 😁. 

I'll explicitly role both the article header and article footer (so your NVDA can skip back/forth to them)

Reply:
Yes, sadly. Just like the history of browser feature compatibility 😰